ambry-frontend/src/main/java/com/github/ambry/frontend/AmbryUrlSigningService.java

```diff
@@ -149,6 +165,20 @@ public String getSignedUrl(RestRequest restRequest) throws RestServiceException
         }
         argsForUrl.put(RestUtils.Headers.CHUNK_UPLOAD, true);
         argsForUrl.put(RestUtils.Headers.SESSION, UUID.randomUUID().toString());
+        BlobId reservedMetadataBlobId = ClusterMapUtils.reserveMetadataBlobId(
```

Comment:
No matter isReservedMetadataEnabled is enabled or disabled, we always generate the reservedMetadataBlobId.
Is this a bug?

Reply:
No this is not a bug. The idea is to do all the while the feature is disabled to collect the metrics. Any user facing changes (like reserved metadata chunk showing up in the signed-url, or exception getting thrown due to reserved metadata chunk), will happen only when the feature is enabled.
Seeing that all metrics look ok will give us enough confidence in a few weeks to enable the feature everywhere.

Reply:
Updated the comments in RouterConfig.